pkg/storage/snapshot/snapshot.go

```diff
@@ -706,6 +708,9 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		vmSnapshotCpy.Status.Phase = snapshotv1.Failed
 		updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, vmSnapshotDeadlineExceededError))
 		updateSnapshotCondition(vmSnapshotCpy, newFailureCondition(corev1.ConditionTrue, vmSnapshotDeadlineExceededError))
+	} else if vmSnapshotError(vmSnapshotCpy) != nil {
```

Comment:
Just out of curiosity, why move this?

Reply:
VM can be in phase progressing while also have an Error, I want this if to happen instead of the progressing if.